app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java

```diff
@@ -389,6 +403,45 @@ private int getClickedGraphicId(float screenX, float screenY) {
         private boolean isEntityClicked(int graphicId) {
             return graphicId != -1;
         }
+
+        private void handleKmlEntityClicks(float screenX, float screenY) {
+            KmlEntityInfo info = getClickedKmlInfo(screenX, screenY);
+            if (info != null) {
+                notifyKmlEntityClicked(info);
+            }
+        }
+
+        private KmlEntityInfo getClickedKmlInfo(float screenX, float screenY) {
+            Set<String> layerIds = mVectorLayerIdToKmlLayerMap.keySet();
```

Comment:
I'm pretty surprised by this... Good to know.
So is this OK?
String[] layerIds = (String[]) mVectorLayerIdToKmlLayerMap.keySet().toArray();